### PR TITLE
Fix error test cases

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
JSON-patch error test cases were tested incorrectly: only first error test case was evaluated.
This pull requests refactors tests to use JUnit `@Parameterized` framework and fixes revealed minor bugs.

### CODE REVIEW

File: src/test/java/com/flipkart/zjsonpatch/AbstractTest.java [hunk 1]
```diff
@@ -1,88 +1,54 @@
...
-//            System.out.println(diff);
-//            Assert.assertThat(diff, equalTo(patch));
-//        }
-//    }
+    private void testOpertaion() throws Exception {
+        JsonNode node = p.getNode();
 
-    @Test(expected = JsonPatchApplicationException.class)
-    public void testErrorsAreCorrectlyReported() {
-        for (int i = 0; i < errorNode.size(); i++) {
```
Comment (AlexanderYastrebov) [inline src/test/java/com/flipkart/zjsonpatch/AbstractTest.java]: Here is the problem with error test cases: once first test case fails with expected exception loop iteration stops and further tests cases are not evaluated

File: src/test/java/com/flipkart/zjsonpatch/AbstractTest.java [hunk 2]
```diff
@@ -1,88 +1,54 @@
...
+        JsonNode second = node.get("expected");
+        JsonNode patch = node.get("op");
+        String message = node.has("message") ? node.get("message").toString() : "";
 
-            System.out.println("Error Test # " + i);
-            System.out.println(first);
-            System.out.println(patch);
+        JsonNode secondPrime = JsonPatch.apply(patch, first);
 
-            JsonNode secondPrime = JsonPatch.apply(patch, first);
```
Comment (AlexanderYastrebov) [inline src/test/java/com/flipkart/zjsonpatch/AbstractTest.java]: Here is the problem with error test cases: once first test case fails with expected exception loop iteration stops and further tests cases are not evaluated
 Add a line note

File: src/test/resources/testdata/rfc6902-samples.json [hunk 3]
```diff
@@ -3,7 +3,8 @@
         {
             "message": "A.13.  Invalid JSON Patch Document",
             "op": [{ "op": "add", "path": "/baz", "value": "qux", "op": "remove" }],
-            "node": { "foo": "bar" }
+            "node": { "foo": "bar" },
+            "disabled": true
```
Comment (AlexanderYastrebov) [inline src/test/resources/testdata/rfc6902-samples.json]: This test case is disabled because duplicate keys are handled by Jackson

**Comment by vishwakarma** [comment]
Thanks for PR

**Pull request merged**